tests/build/test_utils.py

```diff
@@ -108,3 +112,134 @@ def test_process_accelerate_launch_custom_config_file(patch_path_exists):
+        self.source_sub_dir = os.path.join(self.source_dir, "subdir1")
+
+        os.mkdir(self.source_dir)
+        for file_number in range(2):
+            with open(
+                os.path.join(self.source_dir, f"tf{file_number+1}.txt"),
+                "a",
+                encoding="utf-8",
+            ) as f:
+                f.close()
+
+        os.mkdir(self.source_sub_dir)
+        for file_number in range(2):
+            with open(
+                os.path.join(self.source_sub_dir, f"tf{file_number+4}.txt"),
+                "a",
+                encoding="utf-8",
+            ) as f:
+                f.close()
+
+    def are_dir_trees_equal(self, dir1, dir2):
+
+        dirs_cmp = filecmp.dircmp(dir1, dir2)
+        if (
+            len(dirs_cmp.left_only) > 0
+            or len(dirs_cmp.right_only) > 0
+            or len(dirs_cmp.funny_files) > 0
+        ):
+            return False
+        (_, mismatch, errors) = filecmp.cmpfiles(
+            dir1, dir2, dirs_cmp.common_files, shallow=False
+        )
+        if len(mismatch) > 0 or len(errors) > 0:
+            return False
+        for common_dir in dirs_cmp.common_dirs:
+            new_dir1 = os.path.join(dir1, common_dir)
+            new_dir2 = os.path.join(dir2, common_dir)
+            if not self.are_dir_trees_equal(new_dir1, new_dir2):
+                return False
+        return True
+
+
+def test_copy_checkpoint_dest_dir_does_not_exist():
+
+    # Init source directory
+    with tempfile.TemporaryDirectory() as test_root:
+        config = CopyCheckpointTestConfig(test_root)
+
+        target_dir_does_not_exist = os.path.join(
+            config.test_root, "test_copytree_target"
+        )
+
+        # Execute the copy
+        copy_checkpoint(config.source_dir, target_dir_does_not_exist)
+        assert config.are_dir_trees_equal(config.source_dir, target_dir_does_not_exist)
+
+
+def test_copy_checkpoint_dest_dir_does_exist():
+
+    # Init source directory
+    with tempfile.TemporaryDirectory() as test_root:
+        config = CopyCheckpointTestConfig(test_root)
+
+        # Init target directory
+        target_dir_does_exist = os.path.join(config.test_root, "test_copytree_target2")
+        os.mkdir(target_dir_does_exist)
+        # Add a file to the target. This file will be overwritten during the copy.
+        with open(
+            os.path.join(target_dir_does_exist, "tf1.txt"),
+            "a",
+            encoding="utf-8",
+        ) as f:
+            f.close()
+        # Add a file to the target. This file does not exist in source.
+        with open(
+            os.path.join(target_dir_does_exist, "tf9.txt"),
+            "a",
+            encoding="utf-8",
+        ) as f:
+            f.close()
+        # Execute the copy
+        copy_checkpoint(config.source_dir, target_dir_does_exist)
+        # Validate the file that did not exist in source is still there.
```

Comment:
This comment seems to be in conflict with the comment on line 200. So for clarity, if the destination dir already exists and has a file in it, will that file still exist after the copying of source to dest? I think the answer is yes unless it has the same name as a file from source, in which it would be overwritten?

Reply:
@olson-ibm I see you deleted this comment but that doesn't address the issue or my question

Reply:
Ah I once again misunderstood and in this test Joe is doing two things...he is seeing that a file that already exists in target_dir and has same name as a file getting copied over gets overwritten and he is seeing that a file that  already exists in target_dir and has diff name does not get overwritten. So this comment does make sense but is a little hard to follow with the filenames